Assets/Scripts/PlayerController/PlayerMovement.cs

```diff
@@ -70,7 +71,13 @@ void Update()
 
         float t = Time.deltaTime;
         float t2 = t * t;
-        m_CharacterController.Move(new Vector3(m_Velocity.x, m_Velocity.y * t + 0.5f * GetGravityMagnitude() * t2, m_Velocity.z));
+
+        if (m_cannotMove)
+        {
+            m_CharacterController.Move(new Vector3(0, m_Velocity.y * t + 0.5f * GetGravityMagnitude() * t2, 0));
```

Comment:
We can avoid the repeated code by computing the final velocity outside the branch and then use `new Vector3(0, finalVelocity.y, 0)` in one branch and `finalVelocity` in the other.